### PR TITLE
Created a monitor to recover slaves with status offline if is up

### PR DESCRIPTION
**Pull request by thiagohora**
Create a monitor to recover slaves with status offline if is up

### CODE REVIEW

**Comment by ngaut** [comment]
The behavior is dangerous if we switch master and slave many times in a few minutes. Since we may haven't finish the replication between the master and slave. Any thoughts ?

**Comment by thiagohora** [comment]
I understand, but I believe that it should to be verified before the promove command (maybe using SYNC), Given that this feacture only return the node to slot and theoretically the replica would be done before the promove action . What you think?

**Comment by ngaut** [comment]
@thiagohora I think we should check many more states between the new master and slave. Such as some variables in the output of redis command "states".

**Comment by thiagohora** [comment]
Yes, we can verify if the promove is occuring for fail or trigger of dashboard for example, I did it thinking in a fast fail with recover automatic, only to become  the instance that fail eligible again. Also think that we can try a sincronisation or migration during the process of change from offline to slave, before the change state

File: servergroup.go [hunk 1]
```diff
@@ -67,6 +79,19 @@ func handleCrashedServer(s *models.Server) error {
 	return nil
 }
 
+func handleAddServer(s *models.Server) error {
+
```
Comment (ngaut) [inline servergroup.go]: Remove this empty line.

File: servergroup.go [hunk 2]
```diff
@@ -67,6 +79,19 @@ func handleCrashedServer(s *models.Server) error {
...
+	s.Type = models.SERVER_TYPE_SLAVE
+	log.Infof("try return slave %+v", s)
```
Comment (ngaut) [inline servergroup.go]: s/return/reusing/g

Comment (thiagohora) [inline servergroup.go]: Sorry I dont understand, Change s for g is it?

Comment (ngaut) [inline servergroup.go]: It means replace "return" to "reusing"

Comment (thiagohora) [inline servergroup.go]: Ok, done

**Comment by ngaut** [comment]
LGTM. Thanks @thiagohora

**Comment by thiagohora** [comment]
Oh, I'm so happy for can help

**Pull request merged**